web/src/upload.js

```diff
@@ -332,6 +332,10 @@ export function setup_upload(config) {
         if (config.mode === "compose" && !compose_state.composing()) {
             compose_actions.respond_to_message({trigger: "file drop or paste"});
         }
+        // Halt the normal browser paste event, which would otherwise
+        // present a plain-text version of the file name. We do this as
+        // late as possible, before making the call to upload the files
```

Comment:
Why do we do it as late as possible; is the motivation just the `!clipboard_data` return, or something else?

Reply:
It was for the sake of that return, although I discovered today that calling `event.preventDefault()` would invite a serious regression against any pastes for which we don't have explicit handlers--e.g., if someone just wanted to paste a string of vanilla plaintext.

In the updated commit I've pushed, I've introduced an early return when no files are found on the Clipboard. And it's now only after that early return that it's safe to call `event.preventDefault()`.